src/Types.hs

```diff
@@ -110,7 +110,7 @@ data Name =
 data ListWithLength t a = ListWithLength (L.GenericList Name t a) (Maybe Int)
 
 listList :: Lens' (ListWithLength t a) (L.GenericList Name t a)
-listList f (ListWithLength a b) = (\a' -> ListWithLength a' b) <$> f a
```

Comment:
I'd prefer the Lens definitions be left as they were; infix notation is pretty awful here IMO.

So use a `{- ANN -}` module annotation to suppress this warning.  I'm sure there's already some modules where I did that.

Reply:
Ah it can be done per-function, e.g.: 

```haskell
listList = blah blah blah
{-# ANN listList ("HLint: ignore Avoid lambda" :: String) #-}
```

Likewise for the other functions.

Reply:
Ahh yep gotcha. Hopefully fixed now. I did this half-blind, since it seems the new hlint release also needs a higher GHC release than I currently have in my environment. Haven't had time to look into it.